maltoolbox/language/languagegraph.py

```diff
@@ -985,31 +1565,10 @@ def _get_attacks_for_asset_type(self, asset_type: str) -> dict:
             'Get attack steps for %s asset from '
             'language specification.', asset["name"]
         )
-        if asset['superAsset']:
-            logger.debug('Asset extends another one, fetch the superclass '\
-                'attack steps for it.')
-            attack_steps = self._get_attacks_for_asset_type(asset['superAsset'])
 
         for step in asset['attackSteps']:
             if step['name'] not in attack_steps:
                 attack_steps[step['name']] = copy.deepcopy(step)
```

Comment:
Why not do this:

```suggestion
        attack_steps = copy.deepcopy(asset['attackSteps'])
```

Is there ever a case where an lang-graph asset will contain the same step twice? Shouldn't this be considered a bug at the .mal level?

Also, I don't think deepcopy is needed here. The returned `LanguageGraphAttackStep`'s are only used for lookup of step info when constructing the attack graph.

Reply:
No on both counts, I think these were all leftovers from when this function would assemble the inherited steps as well.